src/web-utils/videoplayer.ts

```diff
@@ -287,49 +287,23 @@ export class VideoPlayer {
 
   private _getVideoType(): boolean {
     let ret = false;
-    let vType = '';
     const sUrl: string = this._url ? this._url : '';
     if (sUrl != null && sUrl.length > 0) {
       try {
-        const val = sUrl.substring(sUrl.lastIndexOf('/')).match(/(.*)\.(.*)/);
-        if (val == null) {
-          vType = '';
+        const isHls = sUrl.match(/\.(m3u8)/i);
+        if (isHls) {
+          this._videoType = 'application/x-mpegURL';
+          ret = true;
         } else {
-          const a = sUrl.match(/(.*)\.(.*)/);
-          vType = a != null ? a[2].split('?')[0] : '';
-        }
-        switch (vType) {
```

Comment:
Trading a switch for regex and several if statements replaces one set of complex code for another.

Instead consider more performant perhaps more readable lookup:

``` typescript
const videoTypes: Record<string, string> = {
    'm3u8': 'application/x-mpegURL',
    '': 'video/mp4',
    'mp4': 'video/mp4',
    'webm': 'video/mp4',
    'cmaf': 'video/mp4',
    'cmfv': 'video/mp4'
};

function getVideoType(ext: string): string | null {
    return videoTypes[ext] || null;
}
```

In this example there is now only one piece of code/logic for all current and future video types.

Reply:
@harmonwood 
I pushed a new approach that is more clear and will allow to add more rules and extensions if needed. Let me know what do you think.

Reply:
@harmonwood , let me know what do you think of it

Reply:
Hi @harmonwood, let me know if the new method is good to go or more changes are needed.

Reply:
Hi @harmonwood , did you have time to check this?

Reply:
@harmonwood , any update on this?

Reply:
This is good.